File: app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Event extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'title',
        'description',
        'start_date',
        'end_date',
        'location',
        'category',
        'max_volunteers',
        'creator_id',
        'district_id',
        'status',
        'rejection_reason',
    ];

    protected $attributes = [
        'status' => 'pending',
    ];

    protected $casts = [
        'start_date' => 'datetime',
        'end_date'   => 'datetime',
    ];

    // Relationships
    public function creator()
    {
        return $this->belongsTo(User::class, 'creator_id');
    }

    public function district()
    {
        return $this->belongsTo(District::class);
    }

    // كل مشاركات المتطوعين بالفعالية
    public function volunteerParticipations()
    {
        return $this->hasMany(VolunteerEventParticipation::class);
    }

    // المشاركات اللي تحقق فيها المنظم
    public function verifiedParticipations()
    {
        return $this->hasMany(VolunteerEventParticipation::class, 'verified_by');
    }

    // المشاركات اللي اعتمدها المنظم
    public function approvedParticipations()
    {
        return $this->hasMany(VolunteerEventParticipation::class, 'approved_by');
    }

    public function pledges()
    {
        return $this->hasMany(Pledge::class);
    }

    public function reports()
    {
        return $this->morphMany(Report::class, 'reportable');
    }
}

File: lib/Model/Event.dart
<<<<<<< HEAD
import 'package:impactlyflutter/Model/Category.dart';
import 'package:impactlyflutter/Model/Pledges.dart';

=======
>>>>>>> 825b2bb55dfeb431a16107c04ddf047000640836
class Event {
  int? id;
  String? title;
  String? description;
  String? startDate;
  String? endDate;
  String? location;
  String? category;
  int? maxVolunteers;
  int? creatorId;
  int? districtId;
  String? status;
  var rejectionReason;
  String? createdAt;
  String? updatedAt;
  var deletedAt;
  int? volunteerRegistrationsCount;
  String? districtName;
  String? governorateName;
  String? organizerName;
  bool? isRegistered;
  int? registrationId;
<<<<<<< HEAD
  List<Pledges>? requestedPledges;
=======
>>>>>>> 825b2bb55dfeb431a16107c04ddf047000640836

  Event({
    this.id,
    this.title,
    this.description,
    this.startDate,
    this.endDate,
    this.location,
    this.category,
    this.maxVolunteers,
    this.creatorId,
    this.districtId,
    this.status,
    this.rejectionReason,
    this.createdAt,
    this.updatedAt,
    this.deletedAt,
    this.volunteerRegistrationsCount,
    this.districtName,
    this.governorateName,
    this.organizerName,
    this.isRegistered,
    this.registrationId,
<<<<<<< HEAD
    this.requestedPledges,
=======
>>>>>>> 825b2bb55dfeb431a16107c04ddf047000640836
  });

  Event.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    title = json['title'];
    description = json['description'];
    startDate = json['start_date'];
    endDate = json['end_date'];
    location = json['location'];
    category = json['category'];
    maxVolunteers = json['max_volunteers'];
    creatorId = json['creator_id'];
    districtId = json['district_id'];
    status = json['status'];
    rejectionReason = json['rejection_reason'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
    deletedAt = json['deleted_at'];

    if (json.containsKey('volunteer_registrations_count')) {
      volunteerRegistrationsCount = json['volunteer_registrations_count'];
    }
    if (json.containsKey('district_name')) {
      districtName = json['district_name'];
    }
    if (json.containsKey('governorate_name')) {
      governorateName = json['governorate_name'];
    }
    if (json.containsKey('organizer_name')) {
      organizerName = json['organizer_name'];
    }
    if (json.containsKey('is_registered')) {
      isRegistered = json['is_registered'] == true;
    } else {
      isRegistered = false;
    }
    if (json.containsKey('registration_id')) {
      registrationId = json['registration_id'];
    }
<<<<<<< HEAD
    if (json['requested_pledges'] != null) {
      requestedPledges = <Pledges>[];
      json['requested_pledges'].forEach((v) {
        requestedPledges!.add(new Pledges.fromJson(v));
      });
    }
  }

  Event.fromJsonPledge(Map<String, dynamic> json) {
    id = json['id'];
    title = json['title'];
    description = json['description'];
    startDate = json['start_date'];
    endDate = json['end_date'];
    location = json['location'];
    category =
        json['category'] != null
            ? Category.fromJson(json['category']).name
            : null;
    maxVolunteers = json['max_volunteers'];
    creatorId = json['creator_id'];
    districtId = json['district_id'];
    status = json['status'];
    rejectionReason = json['rejection_reason'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
    deletedAt = json['deleted_at'];

    if (json.containsKey('volunteer_registrations_count')) {
      volunteerRegistrationsCount = json['volunteer_registrations_count'];
    }
    if (json.containsKey('district_name')) {
      districtName = json['district_name'];
    }
    if (json.containsKey('governorate_name')) {
      governorateName = json['governorate_name'];
    }
    if (json.containsKey('organizer_name')) {
      organizerName = json['organizer_name'];
    }
    if (json.containsKey('is_registered')) {
      isRegistered = json['is_registered'] == true;
    } else {
      isRegistered = false;
    }
    if (json.containsKey('registration_id')) {
      registrationId = json['registration_id'];
    }
    if (json['requested_pledges'] != null) {
      requestedPledges = <Pledges>[];
      json['requested_pledges'].forEach((v) {
        requestedPledges!.add(new Pledges.fromJson(v));
      });
    }
=======
>>>>>>> 825b2bb55dfeb431a16107c04ddf047000640836
  }
}
